src/main/java/de/rabea/Game/Board.java

```diff
@@ -23,11 +23,25 @@ public Board(Cell[] cells) {
         return cells;
     }
 
-    public Cell[] placeMark(int position, Cell mark) {
-        cells[position] = mark;
+    public Cell[] placeMark(int position, Mark mark) {
+        cells[position] = mark.convertToCell();
         return cells;
     }
 
+    public boolean gameOver() {
+        return isFull() || hasWinner();
+    }
+
+    public boolean hasWinner() {
+        List<List<Integer>> winningCombinations = getAllLines();
+        for (List<Integer> combination : winningCombinations) {
+            if (anyLineMatchesAWinningCombination(cells(), combination)) {
```

Comment:
I'd like you to try to convert this to a stream: https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html